preditor/gui/loggerwindow.py

```diff
@@ -539,41 +543,72 @@ def handleMenuHovered(self, action):
         menu = action.parentWidget()
         QToolTip.showText(QCursor.pos(), text, menu)
 
-    def findCurrentFontAction(self):
-        """Find and return current font's action"""
-        actions = self.uiMonospaceFontMENU.actions()
-        actions.extend(self.uiProportionalFontMENU.actions())
-
-        action = None
-        for act in actions:
-            if act.isChecked():
-                action = act
-                break
-
-        return action
-
-    def selectFont(self, action):
-        """Set console and workbox font to current font
+    def selectFont(self, monospace=False, proportional=False):
+        """Present a QFontChooser dialog, offering, monospace, proportional, or all
+        fonts, based on user choice. If a font is chosen, set it on the console and
+        workboxes.
 
         Args:
-            action (QAction): menu action associated with chosen font
+            action (QAction): menu action associated with chosenO font
         """
+        origFont = self.console().font()
+        curFontFamily = origFont.family()
+
+        if monospace and proportional:
+            options = QFontDialog.MonospacedFonts | QFontDialog.ProportionalFonts
+            kind = "monospace or proportional "
+        elif monospace:
+            options = QFontDialog.MonospacedFonts
+            kind = "monospace "
+        elif proportional:
+            options = QFontDialog.ProportionalFonts
+            kind = "proportional "
+
+        # Present a QFontDialog for user to choose a font
+        title = "Pick a {} font. Current font is: {}".format(kind, curFontFamily)
+        newFont, okClicked = QFontDialog.getFont(
+            origFont, self, title, options=options
+        )
 
-        actions = self.uiMonospaceFontMENU.actions()
-        actions.extend(self.uiProportionalFontMENU.actions())
+        if okClicked:
+            self.console().setConsoleFont(newFont)
+            self.setWorkboxFontBasedOnConsole()
+            self.setEditorChooserFontBasedOnConsole()
 
-        for act in actions:
-            act.setChecked(act == action)
+    def setFontSize(self, newSize):
```

Comment:
I would prefer to have a single unified location to set the font that updates the font size of all workboxes not just the current one. See this attempt I setup on this [branch](https://github.com/blurstudio/PrEditor/commit/fb1fc048c1dd1b983441245295b02d86cc9e97bb#diff-8b07be49be7db026b4f521e983961298a68990fb621a0d3270ab22496bbff4cdR1002).

Reply:
Hmm, yes, I originally had it set all workbox fonts, though I found that when `ctrl+scrolling` (to adjust font size), it got somewhat laggy. That's why I made it onlychange the current workbox, then when any workbox gets focus, it checks if font is not current and updates it. I loaded some large bits of code into workboxes, and it was quick to update when tab got focus. Though the lagginess might have be caused by something else. I'll revisit this.